crates/node/src/interactive_cli/context.rs

```diff
@@ -165,44 +179,84 @@ impl ContextCommand {
             Commands::Join {
                 private_key,
                 invitation_payload,
+                context,
+                identity,
             } => {
+                let public_key = private_key.public_key();
+
                 let response = node
                     .ctx_manager
                     .join_context(private_key, invitation_payload)
                     .await?;
 
-                if let Some((context_id, identity)) = response {
+                if let Some((context_id, public_key)) = response {
+                    // Create identity alias in the context if --name is provided
+                    if let (Some(identity)) = (identity.as_ref()) {
+                        if let Err(e) =
+                            node.ctx_manager
+                                .create_alias(identity, Some(context_id), public_key)
+                        {
+                            eprintln!(
+                                "{} Failed to create identity alias '{}': {e}",
+                                ind,
+                                pretty_alias(context.cyan(), context_id)
+                            );
+                        } else {
+                            println!(
+                                "{} Created identity alias '{}' for {} in context {}",
+                                ind,
+                                identity.cyan(),
+                                public_key.cyan(),
+                                pretty_alias(context.cyan(), context_id)
```

Comment:
I don't think this works, since `pretty_alias` expects an `Option<Alias<T>>`

you'd either have to move the `.cyan()` out, wrapping the output of `pretty_alias`

```suggestion
                                pretty_alias(context, context_id).cyan(),
```

OR, and perhaps preferably, move the `.cyan()` inside `pretty_alias` itself, so the returned string is already colored (without also painting the parenthesis `(` `)`)